### ISSUE TITLE
Question about sequence track output logits

### ISSUE THREAD

**Issue by junhaobearxiong** [evolutionaryscale/esm#46]
Hi! Thank you for this great work! 

I have a question about the outputs of the sequence track. I noticed that the output logits of the sequence track is of shape `[N, L, 64]` (defined [here](https://github.com/evolutionaryscale/esm/blob/95e3c5be8acda407414810ff3aa7d27dbb6e30d3/esm/models/esm3.py#L156)), whereas the vocab size of `EsmSequenceTokenizer` is 33. As far as I can tell, for sampling, the model directly samples tokens from the raw logits of shape `[N, L, 64]`, then decode the tokens back to string using `EsmSequenceTokenizer`. Is there a particular reason for the difference between the output logits shape and the vocab size, or is there something I'm missing?

Thank you so much!

Best,
Bear

**Comment by scandido** [comment]
The shape of the prediction head that computes logits is larger than the actual vocabulary for pretraining efficiency reasons. One can safely ignore the positions in the output that do not correspond to a token in the vocabulary.
